src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8922,4 +8922,18 @@ public static class Extension
             Diagnostic(ErrorCode.ERR_NoImplicitConv, "1").WithArguments("int", "int[]").WithLocation(9, 27)
             );
     }
+
+    [Fact]
+    public void NotExhaustive_LongList()
+    {
+        string source = """
+            var a = new[] { 1, 2, 3 };
+            _ = a switch { { Length: < 1000 } => 0 };
```

Comment:
>Length: < 1000 [](http://example.com/codeflow?start=29&length=14)

Please consider including tests at the limit between the two types of pattern examples:
```csharp
var a = new[] { 1, 2, 3 };
_ = a switch { { Length: not 0 } => 0 };
_ = a switch { { Length: not 1 } => 0 };
_ = a switch { { Length: not 2 } => 0 };
_ = a switch { { Length: not 3 } => 0 };
_ = a switch { { Length: not 4 } => 0 };
```

Reply:
Perhaps also include an equivalent test using list patterns:
```csharp
var a = new[] { 1, 2, 3 };
_ = a switch { not [] => 0 };
_ = a switch { not [_] => 0 };
_ = a switch { not [_, _] => 0 };
_ = a switch { not [_, _, _] => 0 };
_ = a switch { not [_, _, _, _] => 0 };
```